### PR TITLE
Add a page on OpenCue versioning and compatibility.

### LINKED ISSUE #178: Document project versioning policy
Define our policy on how we create version numbers for OpenCue -- conditions under which we will increment major vs. minor vs. patch number.

### PR DESCRIPTION
**Pull request by bcipriano**
## Related issue
Fixes #178 

## Summarize your changes
Define how version numbers are incremented and outline some basic expectations for compatibility between versions of OpenCue.

### CODE REVIEW

File: content/docs/Concepts/versioning.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+---
+title: "Versioning"
+linkTitle: "Versioning"
+weight: 3
+date: 2020-04-10
+description: >
+  OpenCue policy on version number changes and compatibility between versions.
+---
+
+This page defines OpenCue policy on version number changes and compatibility between versions.
```
Comment (sharifsalah) [inline content/docs/Concepts/versioning.md:10]: This page defines **the** OpenCue

File: content/docs/Concepts/versioning.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+---
+title: "Versioning"
+linkTitle: "Versioning"
+weight: 3
+date: 2020-04-10
+description: >
+  OpenCue policy on version number changes and compatibility between versions.
```
Comment (sharifsalah) [inline content/docs/Concepts/versioning.md:7]: Nit: remove the period for the description

File: content/docs/Concepts/versioning.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+---
+
+This page defines OpenCue policy on version number changes and compatibility between versions.
+
+OpenCue version numbers contain three components: `<major>.<minor>.<patch>`.
+
+- Major version will be incremented only be agreement of TSC. Major versions will be rare and
```
Comment (sharifsalah) [inline content/docs/Concepts/versioning.md:14]: Major version will be incremented only **by** agreement of the technical steering committee (TSC).

(Fix a typo and expand acronym on first use)

File: content/docs/Concepts/versioning.md (hunk 1 above)
Comment (bcipriano) [reply content/docs/Concepts/versioning.md]: Done.

File: content/docs/Concepts/versioning.md (hunk 2 above)
Comment (bcipriano) [reply content/docs/Concepts/versioning.md]: Done.

File: content/docs/Concepts/versioning.md (hunk 3 above)
Comment (bcipriano) [reply content/docs/Concepts/versioning.md]: Done.

**Pull request merged**